File: java/leetcode-practice/src/test/java/com/mamta/leetcode/Leet125.java
package com.mamta.leetcode;

import org.junit.jupiter.api.Test;

public class Leet125 {

    public boolean isPalindrome(String s) {

        s = s.replaceAll("[^A-Za-z0-9]", "").toLowerCase();

        int left = 0;
        int right = s.length()-1;

        while(left < right) {
            if(s.charAt(left) != s.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        return true;

    }

    @Test
    public void solution() {
        String s = "A man, a plan, a canal: Panama";
        boolean isPalindrome = isPalindrome(s);

        System.out.println("Is Palindrome: " + isPalindrome);
    }
}

File: python/leet125.py
import re


class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = re.sub(r'[^a-zA-Z0-9]', '', s).lower()

        left = 0
        right = len(s)-1

        while(left<right):
            if s[left] != s[right]:
                return False
            left+=1
            right-=1
        return  True





if __name__ == '__main__':
    s = Solution()
    content = "A man, a plan, a canal: Panama"
    print(s.isPalindrome(content))